sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -864,10 +864,6 @@ def validate(self, validator):
         else:
           setattr(self, 'temp_location', default_bucket)
 
-      if getattr(self, 'staging_location',
-                 None) or getattr(self, 'temp_location', None) is None:
-        errors.extend(validator.validate_gcs_path(self, 'staging_location'))
```

Comment:
For dataflow, staging location has to be GCS. Does this remove the validation? Other runners may have other validation. Should live in the runner.

Reply:
Updates to follow https://cloud.google.com/dataflow/docs/reference/pipeline-options.
Previously it seems we would accept a local filepath for staging_location. See sdks/python/apache_beam/runners/dataflow/template_runner_test.py which depends on writing locally. I updated that as well.

Reply:
I'm not familiar with that runner or test. But obvs a Dataflow worker cannot fetch staged files from a local tmpdir. Sounds like there is some inappropriate implementation inheritance going on maybe?